src/Slot.ts

```diff
@@ -22,9 +48,12 @@ export interface Slot<RequestData=null, ResponseData=void> {
     // Make the Slot callable: this is how an event is triggered
     // TODO: Find a solution to make it possible to omit the requestData as
     // optional only when explicitly typed as such by the client.
-    (requestData: RequestData): Promise<ResponseData>
-    on: (handler: Handler<RequestData, ResponseData>) => Unsubscribe
-    lazy: (connect: () => void, disconnect: () => void) => Unsubscribe
+    (firstArg: string | RequestData, secondArg?: RequestData): Promise<ResponseData>
```

Comment:
At the very least you should make this type safe by adding a type parameter to `Slot`, and correctly specifying the overloads:

```typescript
export interface Slot<RequestData = null, ResponseData = void, Param extends string = never> {
    (param: Param, requestData: RequestData): Promise<ResponseData>
    (requestData: RequestData): Promise<ResponseData>
    on(handler: Handler<RequestData, ResponseData>): Unsubscribe
    on(param: Param, handler: Handler<RequestData, ResponseData>): Unsubscribe
}
```

Docstrings on the different overloads wouldn't hurt either.

Reply:
Yes absolutely! I added the overloads on the method but not on the slot signature.